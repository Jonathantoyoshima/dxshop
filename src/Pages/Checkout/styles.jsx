import styled from 'styled-components';

export const List = styled.div``;

export const Title = styled.h1`
	font-size: 2rem;
	text-align: center;
	margin-top: 16px;
	margin-bottom: 16px;
	text-transform: uppercase;
`;
