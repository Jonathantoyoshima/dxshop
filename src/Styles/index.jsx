import styled from 'styled-components';

export const theme = {
	typography: {
		normal: `font-family: Arial, sans; letter-spacing: 0.1rem;`,
	},
	color: {
		basic: {
			black: '#0c0c0c',
			white: '#fff',
		},
		primary: {
			main: '#4f6f8c',
			light: '#b4c4d9',
			dark: '#254559',
		},
		secondary: {
			light: '#4D4D99',
			dark: '#011c26',
		},
	},
};

export const General = styled.div`
	${(props) => props.theme.typography.normal};
`;

export const Container = styled.div`
	position: relative;
	width: 100%;
	max-width: 1080px;
	padding: 0 16px;
`;

export const Wrapper = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	color: ${(props) => props.theme.color.basic.white};
	background-color: ${(props) =>
		props.theme.color.secondary.dark};
`;

export const Button = styled.button`
	padding: 10px 30px;
	text-transform: uppercase;
	letter-spacing: 0.3rem;
	border-radius: 10px;
	background-color: ${(props) =>
		props.theme.color.primary.dark};
	border: 1px solid
		${(props) => props.theme.color.basic.white};
	color: ${(props) => props.theme.color.basic.white};
	cursor: pointer;
	transition: all 1s ease;
	:hover {
		background-color: ${(props) =>
			props.theme.color.primary.light};
		border: 1px solid
			${(props) => props.theme.color.primary.dark};
		color: ${(props) => props.theme.color.primary.dark};
	}
`;

export const Input = styled.input`
	padding: 8px;
	width: 100%;
	box-shadow: inset -5px -5px 20px 0px rgba(0, 0, 0, 0.5);
`;
