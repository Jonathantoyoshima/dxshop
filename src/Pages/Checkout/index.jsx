import { Button, Container, Wrapper } from 'Styles';
import { List, Title, Display } from './styles';
import CheckItem from 'Components/CheckItem';
import Header from 'Components/Header';
import Total from 'Components/Total';
import IconButton from 'Components/IconButton';
import ArrowLeft from 'Images/icon/arrow-left.svg';

const Checkout = () => {
  return (
    <Wrapper>
      <Header>
        <IconButton src={ArrowLeft} to={'/'} alt={'button-back'} />
        <Container>Logo</Container>
      </Header>
      <Container>
        <Title>Finalizar pedido</Title>
        <List>
          <CheckItem />
          <CheckItem />
          <CheckItem />
          <CheckItem />
        </List>
        <Total />
        <Display>
          <Button>Finalizar Compra</Button>
        </Display>
      </Container>
    </Wrapper>
  );
};

export default Checkout;
