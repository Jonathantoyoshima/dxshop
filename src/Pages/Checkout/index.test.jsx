import React from 'react';
import { ThemeProvider } from 'styled-components';
import { shallow } from 'enzyme';
import { theme } from 'Styles';
import Checkout from './index';

test('Render Build Checkout Snapshot', () => {
  const component = shallow(
    <ThemeProvider theme={theme}>
      <Checkout />
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot();
});
