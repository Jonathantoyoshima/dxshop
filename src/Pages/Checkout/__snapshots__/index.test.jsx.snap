// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Build Checkout Snapshot 1`] = `ShallowWrapper {}`;
